### PR TITLE
ansible-validate-modules part 4: Enable check on ansible-modules-core

### PR DESCRIPTION
**Pull request by gundalow**
##### ISSUE TYPE
- Bugfix Pull Request
##### COMPONENT NAME

multiple
##### ANSIBLE VERSION

```
ansible-playbook --version
ansible-playbook 2.1.0 (travisdebbuild 0f92d581ca) last updated 2016/04/12 21:18:08 (GMT +100)
  lib/ansible/modules/core: (detached HEAD cf01087a30) last updated 2016/04/06 10:14:13 (GMT +100)
  lib/ansible/modules/extras: (detached HEAD 204b4bab56) last updated 2016/04/06 10:14:31 (GMT +100)
  config file = /home/johnb/.ansible.cfg
  configured module search path = Default w/o overrides

```
##### SUMMARY

Part 4 of getting ansible-validate-modules clean on ansible-modules-core. Currently it's only enabled on ansible-modules-extras.

This also fixes:

`make webdocs`

Fix the last issues that have appeared and enable. I don't see any existing PR to fix these in https://ansible.sivel.net/pr/byfile.html

I've removed trailing whitespace on the files that I've edited.

---

This change is [Reviewable](https://reviewable.io/reviews/ansible/ansible-modules-core/3526)

### CODE REVIEW

**Comment by gundalow** [comment]
@sivel Just a heads up to let you know this fixes the last few issues and enables your `ansible-validate-modules` tool on ansible-modules-extras

**Comment by gregdek** [comment]
Thanks @gundalow. To the current maintainers, @emonty, @shrews, @juliakreger, @j2sol, @rcarrillocruz, @bpennypacker please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit', 'needs_revision' or 'close_me' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gundalow** [comment]
DELETED INVALID COMMENT THAT WAS HERE

**Comment by gregdek** [comment]
Thanks @gundalow for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gundalow** [comment]
Argh bit rot, various new/updated modules that fail this validation. I guess that proves that my change is useful though.

I'll get that fixed up now.

**Comment by gundalow** [comment]
ready_for_review

File: cloud/docker/docker_image_facts.py [hunk 1]
```diff
@@ -111,7 +111,7 @@
                 "Cmd": [
                     "/bin/sh",
                     "-c",
-                    "#(nop) CMD [\"/etc/docker/registry/config.yml\"]"
+                    "#(nop) CMD ['/etc/docker/registry/config.yml']"
```
Comment (agaffney) [inline cloud/docker/docker_image_facts.py]: According to the upstream Dockerfile reference, this is parsed as JSON and _must_ use double-quotes.

https://docs.docker.com/engine/reference/builder/#cmd

**Comment by gregdek** [comment]
Thanks @gundalow. To the current maintainers, @emonty, @shrews, @juliakreger, @j2sol, @rcarrillocruz, @bpennypacker, @privateip please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit', 'needs_revision' or 'close_me' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gundalow** [comment]
needs_revision as stated by @agaffney

**Comment by gundalow** [comment]
@agaffney Good spot, thanks.

ready_for_review

**Comment by abadger** [comment]
Merged to devel and stable-2.1 branches.

**Pull request merged**